packages/features/ee/teams/lib/removeMember.ts

```diff
@@ -101,16 +113,91 @@ const removeMember = async ({
       }),
       ProfileRepository.delete({
         userId: membership.userId,
-        organizationId: orgInfo.id,
+        organizationId: team.id,
       }),
     ]);
+
+    // delete all workflowReminders of the removed team member that come from org workflows
+    const workflowRemindersToDelete = await prisma.workflowReminder.findMany({
+      where: {
+        workflowStep: {
+          workflow: {
+            teamId: team.id,
+          },
+        },
+        booking: {
+          eventType: {
+            userId: memberId,
+          },
+        },
+      },
+      select: {
+        id: true,
+        referenceId: true,
+        method: true,
+      },
+    });
+
+    deleteAllWorkflowReminders(workflowRemindersToDelete);
   }
 
   // Deleted managed event types from this team from this member
   await prisma.eventType.deleteMany({
     where: { parent: { teamId: teamId }, userId: membership.userId },
   });
 
+  // cancel/delete all workflowReminders of the removed member that come from that team (org teams only)
+  if (team.parentId) {
+    const isUserActiveOnOtherTeam = !!foundUser.teams.filter(
+      (userTeam) => userTeam.team.id !== team.id && !!userTeam.team.parentId
+    ).length;
+
+    const removedWorkflows = await prisma.workflow.findMany({
+      where: {
+        activeOnTeams: {
+          some: {
+            teamId: team.id,
+          },
+          //don't delete reminder, if user is still part of another team that is active on this workflow
+          none: {
+            team: {
+              members: {
+                some: {
+                  userId: memberId,
+                },
+              },
+            },
+          },
+        },
+        ...(isUserActiveOnOtherTeam && {
+          isActiveOnAll: false,
+        }),
```

Comment:
I don't get this logic

Reply:
I added comments to the code to explain, also found another case that wasn't covered: https://github.com/calcom/cal.com/pull/15083/files#r1638549350